src/ngraph/distributed/open_mpi.hpp

```diff
@@ -37,20 +37,21 @@ namespace ngraph
             {
                 int flag = 0;
                 MPI_Initialized(&flag);
-                if (!flag)
+                if (!flag && !m_initialized_mpi)
                 {
                     MPI_Init(NULL, NULL);
+                    m_initialized_mpi = true;
                 }
             }
 
             ~OpenMPIDistributedInterface() override
             {
-                int flag = 0;
-                MPI_Initialized(&flag);
-
-                if (flag)
+                int is_mpi_finalized = 0;
+                MPI_Finalized(&is_mpi_finalized);
```

Comment:
if this condition possible:
`is_mpi_finalized && m_initialized_mpi == True`

If yes, could we add a warning here? I don't expect this condition to be true but if it is, we want to know.